walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,66 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
+/** Finds potential meeting times. */
 public final class FindMeetingQuery {
+
+  /**
+   * Returns available TimeRanges for the {@code request} given {@code events}, a Collection of
+   * all events so far. 
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> times = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+    
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      // If the event is longer than a day, no time slots are available to book a meeting.
+      return times; 
+    } else if (attendees.size() == 0) {
```

Comment:
no need for 'else if' can just be 'if' given the previous statement returns.

Reply:
Done!